src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1182 @@
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param register_name: name to register the dataset with on TileDB Cloud,
+        defaults to None
+    :param search_uri: URI to search for geospatial dataset files, defaults to None
+    :param pattern: Unix shell style pattern to match when searching for files,
+        defaults to None
+    :param ignore: Unix shell style pattern to ignore when searching for files,
+        defaults to None
+    :param dataset_list_uri: URI with a list of dataset URIs, defaults to None
+    :param max_files: maximum number of URIs to read/find,
+        defaults to None (no limit)
+    :param compression_filter: serialized tiledb filter,
+        defaults to None
+    :param workers: number of workers for dataset ingestion, defaults to MAX_WORKERS
+    :param batch_size: batch size for dataset ingestion, defaults to BATCH_SIZE
+    :param tile_size: for rasters this is the tile (block) size
+        for the merged destination array, defaults to 1024
+    :param chunk_size: for point cloud this is the PDAL chunk size, defaults to 1000000
+    :param nodata: NODATA value for raster merging
+    :param resampling: string, resampling method,
+        one of None, bilinear, cubic, nearest and average
+    :param res: Tuple[float, float], output resolution in x/y
+    :param verbose: verbose logging, defaults to False
+    :param trace: bool, enabling log tracing, defaults to False
+    :param log_uri: log array URI
+    :param unit_testing: bool internal flag for unit testing the code flow
+    """
+
+    logger = get_logger_wrapper(verbose)
+
+    kwargs = {}
+    fn = None
+
+    if dataset_list_uri:
+        sources = read_uris(
+            dataset_list_uri,
+            dataset_type=dataset_type,
+            log_uri=log_uri,
+            config=config,
+            max_files=max_files,
+        )
+
+    def pointcloud_match(f):
+        return os.path.splitext(f)[-1].lstrip(".").lower() in ["las", "laz"]
+
+    def raster_match(f):
+        try:
+            rasterio.driver_from_extension(f)
+            return True
+        except ValueError:
+            return False
+
+    def geometry_match(f):
+        try:
+            fiona.drvsupport.driver_from_extension(f)
+            return True
+        except ValueError:
+            return False
+
+    if dataset_type == "pointcloud":
+        fn = ingest_point_cloud_udf
+
+        if search_uri:
+            if pattern:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    excludes=ignore,
+                    includes=pattern,
+                    max_files=max_files,
+                )
+            else:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    excludes=ignore,
+                    includes=pointcloud_match,
+                    max_files=max_files,
+                )
```

Comment:
You could shorten this with the following:
```python
sources = find(
                    search_uri,
                    config=config,
                    excludes=ignore,
                    includes=pattern if pattern else pointcloud_match,
                    max_files=max_files,
                )
```

Reply:
👍